tests/integration_tests/ui_tests/ArelleGUITest/ArelleGUITest/Tests.cs

```diff
@@ -0,0 +1,362 @@
+using FlaUI.Core;
+using FlaUI.Core.AutomationElements;
+using FlaUI.Core.Definitions;
+using FlaUI.Core.Input;
+using FlaUI.Core.Logging;
+using FlaUI.Core.Tools;
+using FlaUI.Core.WindowsAPI;
+using FlaUI.TestUtilities;
+using FlaUI.UIA3;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO.Compression;
+using System.Text;
+
+namespace ArelleGUITest
+{
+    [TestFixture]
+    public class ArelleGUITests : FlaUITestBase
+    {
+        protected override ApplicationStartMode ApplicationStartMode => ApplicationStartMode.OncePerTest;
+        protected override VideoRecordingMode VideoRecordingMode => VideoRecordingMode.OnePerTest;
+
+        abstract record ExecutionTypeT(string? ArellePath, string ArelleResourcesPath)
+        {
+            public record Source(string? ArellePath, string ArelleResourcesPath, bool UseVirtualEnv, string PythonExe)
+                : ExecutionTypeT(ArellePath, ArelleResourcesPath);
+            public record Build(string? ArellePath, string ArelleResourcesPath) : ExecutionTypeT(ArellePath, ArelleResourcesPath);
+        }
+
+        readonly ExecutionTypeT ExecutionType;
+        Window Window;
+
+        public ArelleGUITests()
+        {
+            var useSource = Environment.GetEnvironmentVariable("ARELLE_USE_BUILD") != "true";

```

Comment:
Any reason to not use strong type like `boolean`?

Reply:
Sure, but its weakly typed using a `var` instead of strongly typed using `bool useSource = ...;`

Reply:
Sorry, wrong phrasing here, the point is the same though, why use `var` instead of `bool`. According to the [docs](http://msdn.microsoft.com/en-us/library/bb384061.aspx):
> The var keyword can also be useful when the specific type of the variable is tedious to type on the keyboard, or is obvious, or does not add to the readability of the code. One example where var is helpful in this manner is with nested generic types such as those used with group operations. In the following query, the type of the query variable is IEnumerable>. As long as you and others who must maintain your code understand this, there is no problem with using implicit typing for convenience and brevity.

> However, the use of var does have at least the potential to make your code more difficult to understand for other developers. For that reason, the C# documentation generally uses var only when it is required.

explicit types should be preferred when possible to make the code less difficult to understand

Reply:
🤷 it seems like an interesting decision to go against the guidance provided by the language maintainers but that said I suppose I also could have picked a better one to comment on. Here is an example of a place where I would have to execute/debug the code/have the api up to figure out what type I am expecting to be returned here :

```cs
var iconsAndMainArea = Window.FindFirstByXPath("/Pane/Pane");
var iconsAndMainAreaChildren = iconsAndMainArea.FindAllChildren();
var mainArea = iconsAndMainAreaChildren[0];
var icons = iconsAndMainAreaChildren[1];
var statusBar = iconsAndMainAreaChildren[2];
var mainAreaChildren = mainArea.FindAllChildren();
var messagesAreaAndTabBar = mainAreaChildren[0];
var tablesAndFactView = mainAreaChildren[1];
var messagesArea = messagesAreaAndTabBar.FindFirstChild();
var tablesAndFactViewChildren = tablesAndFactView.FindAllChildren();
var factViewAndTabBar = tablesAndFactViewChildren[0];
var tablesAndTabBar = tablesAndFactViewChildren[1];
var factView = factViewAndTabBar.FindFirstChild();
var tables = tablesAndTabBar.FindFirstChild();
```